### ISSUE TITLE
TypeScript: Synchronous effects' parameters requires Action<any, void>

### ISSUE THREAD

**Issue by zsxsoft** [rematch/rematch#520]
Same issue with: https://github.com/rematch/rematch/issues/478

Reproduce: 

1. Add a synchronous effect in https://github.com/rematch/rematch/blob/master/examples/ts/count/src/models/dolphins.ts.

```typescript
	effects: dispatch => ({
		aaa (s: string) {
			// do nothing
		},
		async incrementAsync(a: number) {
			await delay(500)
			dispatch.dolphins.increment()
		},
	}),
```

2. Call it:
![image](https://user-images.githubusercontent.com/2398785/45866781-6e793400-bdb4-11e8-9724-02d0ed4e15be.png)

TypeScript returns:
```
[ts] Argument of type '111' is not assignable to parameter of type 'Action<any, void>'.

**Comment by ShMcK** [comment]
Issue should be resolved with @rematch/core@1.0.2. Thanks @kkshyu

**Comment by raitucarp** [comment]
this issue still exist on @rematch/core@1.0.6
